### PR TITLE
Removes deprecated log4j

### LINKED ISSUE #6722: Remove log4j logger
sbt 1.x currently has two logger implementations. We should be able to remove log4j one in sbt 2.x?

### LINKED ISSUE #6734: API: Replacement for deprecated LogExchange methods aren't public
Ref https://github.com/sbt/sbt/issues/6722

### Issue

Some methods in `sbt.util.LogExchange` are deprecated since 1.4.0, with the suggestion to use `sbt.util.LoggerContext` instead.

These methods include:
* `LogExchange.logger`
* `LogExchange.unbindLoggerAppenders`
* `LogExchange.bindLoggerAppenders`

It seems that all essential methods of `LoggerContext` which could be used instead are more or less private to `sbt`, `sbt.util` or even `sbt.util.LoggerContext`.

### Context

We use Zinc in Mill as Java/Scala compiler, and in order to get proper compiler messages, e.g. to forward it to BSP clients, we need to be able to create `ManagedLoggedReporter` and as a precondition also `ManagedLogger`. This is what the deprecated API did.

https://github.com/com-lihaoyi/mill/blob/538ffe26f209cff93255f812e0501db4488335aa/scalalib/worker/src/mill/scalalib/worker/ZincWorkerImpl.scala#L416-L443

What is the recommended solution?

### PR DESCRIPTION
**Pull request by Nirvikalpa108**
Fixes https://github.com/sbt/sbt/issues/6722
Fixes https://github.com/sbt/sbt/issues/6734

@eed3si9n and I are working on removing log4j, because the new logger (PR #5731) is working nicely. This will be part of sbt 2.0.0.

### CODE REVIEW

File: internal/util-logging/src/main/scala/sbt/util/LogExchange.scala [hunk 1]
```diff
@@ -29,54 +28,18 @@ sealed abstract class LogExchange {
...
-      .setBlocking(false)
-      .setConfiguration(config)
-      .build
-    asyncAppender.start
-    config.addAppender(asyncAppender)
-    asyncAppender
+  ): Unit = {
+    appenders.map(LoggerContext.globalContext.addAppender(loggerName, _))
+    ()
   }
```
Comment (Nirvikalpa108) [inline internal/util-logging/src/main/scala/sbt/util/LogExchange.scala:43]: Removes the deprecation warnings on the `logger`, `unbindLoggerAppenders` and `bindLoggerAppenders` methods given https://github.com/sbt/sbt/issues/6734.

File: internal/util-logging/src/main/scala/sbt/util/LogExchange.scala [hunk 2]
```diff
@@ -29,54 +28,18 @@ sealed abstract class LogExchange {
     Util.ignoreResult(configs.putIfAbsent(name, config))
   private[util] def removeConfig(name: String): Option[LoggerConfig] = Option(configs.remove(name))
 
-  @deprecated("Use LoggerContext to create loggers", "1.4.0")
   def logger(name: String): ManagedLogger = logger(name, None, None)
-  @deprecated("Use LoggerContext to create loggers", "1.4.0")
   def logger(name: String, channelName: Option[String], execId: Option[String]): ManagedLogger =
     LoggerContext.globalContext.logger(name, channelName, execId)
-  @deprecated("Use LoggerContext to unbind appenders", "1.4.0")
```
Comment (ignasi35) [inline internal/util-logging/src/main/scala/sbt/util/LogExchange.scala:37]: These 3 un-deprecations look weird.

Comment (ignasi35) [reply internal/util-logging/src/main/scala/sbt/util/LogExchange.scala:37]: nevermind, I just saw https://github.com/sbt/sbt/pull/6726/files#r762409111

**Comment by eed3si9n** [comment]
I want to make progress in develop branch, so I'm going to merge this as-is.

**Pull request merged**